spark_auto_mapper/data_types/amount.py

```diff
@@ -23,15 +23,15 @@ def __init__(self, value: AutoMapperAmountInputType):
     def get_column_spec(
         self, source_df: Optional[DataFrame], current_column: Optional[Column]
     ) -> Column:
-        if source_df is not None and isinstance(self.value, AutoMapperDataTypeColumn) \
-                and dict(source_df.dtypes)[self.value.value] not in ("float", "double"):
-            # parse the amount here
+        if isinstance(self.value, AutoMapperDataTypeColumn) \
```

Comment:
Is it clearer to write this like:

```
column_spec = self.value.get_column_spec(
    source_df=source_df, current_column=current_column
)
if <all the if stuff>:
    column_spec = column_spec.cast("double")

return column_spec
```

I think it's equivalent, and is a bit more obvious about what's happening here.

Reply:
Updated.